### ISSUE TITLE
Internal Server Error after Updating Theming App to 1.4.4

### ISSUE THREAD

**Issue by nickian** [nextcloud/server#9595]
### Steps to reproduce
1. Update "Theming" app to 1.4.4
2. Go to Theming settings in Administration

### Expected behaviour
Be able to go to admin, theme settings and modify them.

### Actual behaviour

Internal Server Error

The server was unable to complete your request. If this happens again, please send the technical details below to the server administrator. More details can be found in the server log.

### Server configuration

**Operating system**: Ubuntu 16.04

**Web server:** Apache

**Database:** MySQL

**PHP version:** 7.0

**Nextcloud version:** (see Nextcloud admin page) 13.0.2

#### Web server error log

No errors in Apache log.

#### Nextcloud log (data/nextcloud.log)
```
Error: Call to undefined method OCA\Theming\ThemingDefaults::getImprintUrl()

/www/settings/Controller/CommonSettingsTrait.php - line 115: OCA\Theming\Settings\Admin->getForm()

/www/settings/Controller/AdminSettingsController.php - line 74: OC\Settings\Controller\AdminSettingsController->formatSettings(Array)

/www/settings/Controller/CommonSettingsTrait.php - line 126: OC\Settings\Controller\AdminSettingsController->getSettings('theming')

/www/settings/Controller/AdminSettingsController.php - line 65: OC\Settings\Controller\AdminSettingsController->getIndexResponse('admin', 'theming')
[internal function] OC\Settings\Controller\AdminSettingsController->index('theming')

/www/lib/private/AppFramework/Http/Dispatcher.php - line 161: call_user_func_array(Array, Array)

/www/lib/private/AppFramework/Http/Dispatcher.php - line 91: OC\AppFramework\Http\Dispatcher->executeController(Object(OC\Settings\Controller\AdminSettingsController), 'index')

/www/lib/private/AppFramework/App.php - line 115: OC\AppFramework\Http\Dispatcher->dispatch(Object(OC\Settings\Controller\AdminSettingsController), 'index')

/www/lib/private/AppFramework/Routing/RouteActionHandler.php - line 47: OC\AppFramework\App main('OC\\Settings\\Con...', 'index', Object(OC\AppFramework\DependencyInjection\DIContainer), Array)
[internal function] OC\AppFramework\Routing\RouteActionHandler->__invoke(Array)

/www/lib/private/Route/Router.php - line 297: call_user_func(Object(OC\AppFramework\Routing\RouteActionHandler), Array)

/www/lib/base.php - line 999: OC\Route\Router->match('/settings/admin...')

/www/index.php - line 37: OC handleRequest()

{main}
```

**Comment by MorrisJobke** [comment]
cc @blizzz @juliushaertl

**Comment by juliushaertl** [comment]
I cannot reproduce that error. @nickian Can you try to install the app again by downloading the app store release and replacing the files manually? Maybe there was an issue when replacing the files.

**Comment by blizzz** [comment]
Cannot reproduce it either. How do you run Nextcloud? Using the docker appliance, or different app paths? Is opcache configured and configured correctly (i.e. no warnings on admin page)?

**Comment by Moimemeici** [comment]
I have this problem, 

I'am using Docker

**Comment by blizzz** [comment]
Docker: it might  be related to https://github.com/nextcloud/docker/issues/288 in that case it should work with [next maintenance release](https://github.com/nextcloud/server/pull/9436).

**Comment by nickian** [comment]
I'm not using Docker. I just installed normally by downloading the files. I'll see if I can reinstall the plugin.

**Comment by nickian** [comment]
Yes, must have been an issue that occurred during the update process. When I replaced the Theming files manually it fixed the issue. However, this did happen on 3 different NextCloud installations for me (granted they are on the same server).

**Comment by nickian** [comment]
Perhaps it was a caching issue of some sort. Now the Theming app is working on all installs today (I only replaced the files on one, but they're actually all working now).

**Comment by osos** [comment]
I have the same internal error.

Not sure I understand the fix. Please advise!

**Comment by osos** [comment]
Sorry for the ignorance.

Steps to fix:

1. Goto https://apps.nextcloud.com/apps/theming
2. Download latest (https://github.com/nextcloud/server/files/2037121/theming.tar.gz)
3. untar into /path/to/nextcloud/apps/theming

**Comment by blizzz** [comment]
See my questions from https://github.com/nextcloud/server/issues/9595#issuecomment-392030119 ...abd yes, it sounds like an (op)cache issue. 

Out of curiosity, why didn't you upgrade from within the apps management?

**Comment by nickian** [comment]
@blizzz I did upgrade within apps management.

Actually, my admin/basic settings page did say "No memory cache has been configured." I have since added those lines for APCu and Redis to my config.

That shouldn't matter though. The documentation says: "A memcache is not required and you may safely ignore the warning if you prefer."

**Comment by blizzz** [comment]
@nickian rather misconfigured opcache, but then you'd also see warnings on the admin page.

**Comment by mr-bolle** [comment]
The reason is, that the default app path are not writeable. 
I change `'writable' => false,` to `'writable' => true,` 
into my config/config.php. And try the Update of the app again.

Yes this is the same issue like https://github.com/nextcloud/docker/issues/288

>'apps_paths' =>
  array (
    0 =>
    array (
      'path' => '/var/www/html/apps',
      'url' => '/apps',
      'writable' => false,
    ),

**Comment by michaelarnauts** [comment]
I tried this and changed the writable to `true` on the config, and after a restart, the app still doesn't update. The files in `app` are owned by www-data:root.

There don't seem to be any log entry produced by this.

**Comment by mr-bolle** [comment]
A log entry you don't get, only after refresh the application page the update appears again. If your app path permission into your config.php is set to  'writable' => true, it should works.

**Comment by juliushaertl** [comment]
Closing this in favour of https://github.com/nextcloud/server/issues/8929
